samples/unit-tests/maps/mapline-linewidth/demo.js

```diff
@@ -24,3 +24,45 @@ QUnit.test(
         );
     }
 );
+
+QUnit.test(
+    'Setting lineWidth hover state for the mapLine series (#17105)',
+    function (assert) {
+
+        // Initialize the chart
+        const chart = Highcharts.mapChart('container', {
+            series: [{
+                type: 'mappoint',
+                data: [{
+                    id: 'one',
+                    lat: 1,
+                    lon: 5
+                }, {
+                    id: 'two',
+                    lat: 15,
+                    lon: 1
+                }]
+            }]
+        });
+
+        chart.addSeries({
+            type: 'mapline',
+            states: {
+                hover: {
+                    lineWidth: 4
+                }
+            },
+            data: [{
+                id: 'one - two',
+                path: [["M", 5, 1], ["Q", 2.9268292682926833, 7.8048780487804885, 1, 15]]
+            }]
+        }, true, false);
+
+        chart.series[1].points[0].setState('hover');
+
+        assert.strictEqual(
```

Comment:
We need to test that the setting is actually applied to the SVG as well, not just the options structure. In the [previous test](https://github.com/highcharts/highcharts/blob/251c6e4eb428ff6a317acdf1dcec7944f79224af/samples/unit-tests/maps/mapline-linewidth/demo.js#L17-L24) you can see how to do that.